x-pack/test/functional/apps/security/users.ts

```diff
@@ -111,8 +113,15 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       expect(roles.apm_system.reserved).to.be(true);
       expect(roles.apm_system.deprecated).to.be(false);
 
-      expect(roles.apm_user.reserved).to.be(true);
-      expect(roles.apm_user.deprecated).to.be(true);
+      const esVersion = (await esClient.info()).version.number;
+      log.debug(`Checking ES version: ${esVersion}`);
+
+      if (semver.major(esVersion) < 9) {
+        expect(roles.apm_user.reserved).to.be(true);
+        expect(roles.apm_user.deprecated).to.be(true);
+      } else {
+        log.debug('The `apm_user` role no longer exists in 9.0.0+.');
+      }
```

Comment:
__note:__ The approach was suggested by @marius-dr and looks like the most reliable option we have in tests.